src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -997,9 +1004,28 @@ private TexlNode ParseAs(TexlNode left, ITexlSource leftTrivia)
                 rhsIdentifier);
         }
 
+        private Token GetNewOrExistingToken(Token existingToken)
+        {
+            var isNamedFormulaMode = _flagsMode.Peek().HasFlag(Flags.NamedFormulas);
+            if (existingToken == null || !isNamedFormulaMode || existingToken.Span.BaseIndex != null)

```

Comment:
>existingToken.Span.BaseIndex != null) [](http://example.com/codeflow?start=64&length=37)

Todo: Add comment about this check.

This supposed to be set only once. This is to avoid cloning token twice if baseIndex is already set for this token.